### PR TITLE
feature/snapshot-save

### PR DESCRIPTION
**Pull request by robinfehr**
- ability to save snapshots
- ability to disable snapshot tracking
- refactoring of magic numbers
- introduced atoms
- refactoring of components - mainly made them smarter

### CODE REVIEW

**Comment by Raathigesh** [comment]
First of all thank you so much for an amazing PR 🎉 This would help a lot.

I noticed that once you save a snapshot and then when you switch between other stores or observables, the snapshot you saved from another store or observable still shows up. 
I expect it to only show up on the recorded store and not in other stores.

Is this an intended behavior or a potential bug?

File: packages/app/src/renderer/store/Trackers.js [hunk 1]
```diff
@@ -12,12 +13,26 @@ class Trackers {
...
+        const snapshots = this.savedSnapshots.filter(snapshot =>
+          snapshot.trackerId === this.currentRecordingId
+        );
+        return {
+          snapshots
+        };
+      }),
       currrentTrackerId: 0,
-      trackers: [],
+      trackers: new Map(),
```
Comment (Raathigesh) [inline packages/app/src/renderer/store/Trackers.js]: This is the way to go! I was so careless 👍

**Comment by robinfehr** [comment]
you're right - we should filter the saved snapshots by tracker. will add that.

**Comment by Raathigesh** [comment]
Thank you!

**Comment by Raathigesh** [comment]
Thank you @robinfehr for this PR. Hoping to push a release within a couple of days.

**Comment by robinfehr** [comment]
#1 can be closed I guess - for the persisted version we could open a new issue.

**Pull request merged**